packages/embed/src/utils/setup-google-analytics-instance.ts

```diff
@@ -37,24 +38,32 @@ const logError = (message: string) => {
 }
 
 export const setupGaInstance = (iframe: HTMLIFrameElement, embedId: string, shareGaInstance?: string | boolean) => {
-  try {
-    const gaObject: GoogleAnalyticsObject = window[window.GoogleAnalyticsObject]
-    const trackingId = typeof shareGaInstance === 'string' ? shareGaInstance : undefined
-    const tracker = getTracker(gaObject.getAll(), trackingId)
-
-    if (tracker) {
-      sendGaIdMessage(embedId, tracker.get('clientId'), iframe)
-    } else {
-      logError(
-        `Whoops! You enabled the shareGaInstance feature in your typeform embed but the tracker with ID ${trackingId} was not found. ` +
+  const trackingId = typeof shareGaInstance === 'string' ? shareGaInstance : undefined
+  if (window.gtag) {
+    window.gtag('get', trackingId, 'client_id', (clientId: string) => {
+      sendGaIdMessage(embedId, clientId, iframe)
+    })
```

Comment:
as described here:

https://developers.google.com/tag-platform/gtagjs/reference#get

Reply:
As we only have a callback and have no way of knowing that the gtag('get'...) failed, I added this timeout logic. If the callback is not executed within 3 seconds, I trigger the same log as we were doing for ga